### PR TITLE
170766096 Endpoint to approve trip request

### PR DESCRIPTION
**Pull request by Nigorjeanluc**
#### What does this PR do?
This pull request ensure the user's manager can approve a trip request
### Description of Task to be completed?
 - create `server/middlewares/isManager.js` to verify if the user's role is a manager
 - create `server/middlewares/statusValidation.js` to validate if the trip's status is pending, approved or rejected.
 - create `server/tests/requestTest.js` to this endpoint
#### How should this be manually tested?
You can pull the `develop` or `ft-avail-request-170766096` branch and run these following commands `npm install` and run `npm test` 
#### What are the relevant pivotal tracker stories?
[#170766096](https://www.pivotaltracker.com/n/projects/2429839/stories/170766096)

### CODE REVIEW

File: seeders/20200121095153-demo-user.js [hunk 1]
```diff
@@ -4,15 +4,15 @@ const up = (queryInterface) => queryInterface.bulkInsert('Users', [{
...
   birthdate: new Date(),
   preferredLanguage: 'French',
   preferredCurrency: 'Euro',
-  location: 'Paris',
-  role: 'Broker',
+  locationId: 3,
+  role: 'Requester',
   department: 'Cleaner',
-  lineManager: 'MrNobody',
+  lineManager: 'Runyamahe',
```
Comment (istafrica-dennw) [inline seeders/20200121095153-demo-user.js]: Is a line manager a string or a real user of manager role? work on this with everyone else who is using the same settings.

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]

File: server/controllers/requestController.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  /**
+   * This method approve the trip request.
+   * @param {object} req The user's trip request.
+   * @param {object} res The response.
+   * @returns {object} The status and some data of the trip.
+   */
+  static async approveRequest(req, res) {
+    const reqq = await RequestHelper.findRequest('id', parseInt(req.params.id, 10));
+
+    if (!reqq) {
```
Comment (istafrica-dennw) [inline server/controllers/requestController.js]: if this condition passes, what will happen to line 26? It will still run, a nightmare. fix this

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]

File: seeders/20200121095153-demo-user.js (hunk 1 above)
Comment (Nigorjeanluc) [reply seeders/20200121095153-demo-user.js]: Here for the line manager, I used the username of a user with the role of manager

File: server/controllers/requestController.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      res.status(400).json({
+        status: 400,
+        error: 'Please provide a authentic request'
+      });
+    }
+
+    const manager = await users.userExists('id', parseInt(reqq.userId, 10));
```
Comment (istafrica-dennw) [inline server/controllers/requestController.js]: 1. what is the difference between manager and properManager line 26,27
2. what shows in the code that we are approving. I only see we are updating the status, maybe we are rejecting

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]

File: server/controllers/requestController.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * This class contains all methods
+ * required to approve/reject
+ *  atrip requests.
+ */
+class RequestController {
+  /**
+   * This method approve the trip request.
+   * @param {object} req The user's trip request.
```
Comment (istafrica-dennw) [inline server/controllers/requestController.js]: @Nigorjeanluc  are you sure req stands for trip request? or http request? Think twice. Thank you

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]

**Review by jajajaden01 — APPROVED** [review]

**Review by manzi-guev — APPROVED** [review]

**Review by karengir — APPROVED** [review]

**Review by istafrica-dennw — APPROVED** [review]

**Pull request merged**